### PR TITLE
✨(dashboard) add dashboard homepage

### PR DESCRIPTION
**Pull request by ssorin**
- Introduced templates for the home page, main menu, header, and language selector.
- Added `IndexView` with authentication to home views and corresponding tests. Updated the Makefile to include a data loading target and refined Pipfile.lock.
- Implement localization by adding French translations and configuring locale paths. Refactor templates and include necessary middleware for internationalization. Adjust URL patterns to support language codes and update tests accordingly.
- Added French translations to dashboard
- Added djlint: configured djlint in dashboard/pyproject.yml, added djlint commands (lint + reformat) in Makefile, added djlint to the dashboard github workflow, cleaned up HTML templates with djlint.
- Added language switcher
- Added 404 and 500 templates

### CODE REVIEW

File: src/dashboard/locale/fr/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+msgid "Home"
+msgstr "Accueil"
+
+#: templates/home/cards/consentement.html:13
+msgid "Consent Management"
+msgstr "Gestion du consentement"
+
+#: templates/home/home.html:6
+msgid "Qualicharge dashboard"
+msgstr "Tableau de bord Qualicharge"
```
Comment (jmaupetit) [inline src/dashboard/locale/fr/LC_MESSAGES/django.po:68]:
```suggestion
msgid "QualiCharge dashboard"
msgstr "Tableau de bord QualiCharge"
```

File: src/dashboard/tests/dashboard/setup.py
Comment (jmaupetit) [inline src/dashboard/tests/dashboard/setup.py:1]: Not sure this file is required, also: `setup.py` is a confusing name for pythonista!

**Review by jmaupetit — COMMENTED** [review]
That's great!

Two general remarks:

1. IMO everything related to a Django app should live in the related app's directory. Thus, each app (_e.g._ `apps/home/`) should contain a `templates/` and a `tests/` directory.
2. I have the feeling that you are testing Django features, some tests appears not required. Instead, it would be preferable to test inter-nationalization's integration for example.

Let's talk about this IRL.

File: Makefile [hunk 2]
```diff
@@ -274,6 +275,13 @@ create-dashboard-superuser: ## create dashboard super user
 	@bin/manage createsuperuser --noinput
 .PHONY: create-dashboard-superuser
 
+loaddata-dashboard: ## load data dashboard
```
[the remark this answers is not in the document]
Comment (ssorin) [reply Makefile:278]: ✅

File: Makefile [hunk 3]
```diff
@@ -274,6 +275,13 @@ create-dashboard-superuser: ## create dashboard super user
...
+	@echo "Running dashboard service database engine…"
+	@$(COMPOSE_UP) --wait postgresql
+	@echo "load data dashboard…"#
+	@bin/manage loaddata dashboard/fixtures/dsfr_fixtures.json
+.PHONY: loaddata-dashboard
```
[the remark this answers is not in the document]
Comment (ssorin) [reply Makefile:283]: ✅

File: Makefile [hunk 4]
```diff
@@ -44,6 +44,7 @@ bootstrap: \
   seed-oidc \
   create-api-superuser \
   create-dashboard-superuser \
+  loaddata-dashboard \
```
[the remark this answers is not in the document]
Comment (ssorin) [reply Makefile:47]: ✅

File: src/dashboard/apps/home/apps.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+"""QualiCharge homepage apps base config."""
+
+from django.apps import AppConfig
+
+
+class HomeConfig(AppConfig):
+    """homepage apps config."""
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/home/apps.py:7]: ✅

File: src/dashboard/apps/home/urls.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+"""Qualicharge : homepage app urls."""
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/home/urls.py:1]: ✅

File: src/dashboard/apps/home/views.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+"""Qualicharge dashboard: homepage app views."""
+
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import TemplateView
+
+
+class IndexView(LoginRequiredMixin, TemplateView):
+    """Index view of the homepage."""
+
+    template_name = "home/home.html"
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/apps/home/views.py:10]: ✅

File: src/dashboard/dashboard/fixtures/dsfr_fixtures.json
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/dashboard/fixtures/dsfr_fixtures.json:1]: ✅

File: src/dashboard/locale/fr/LC_MESSAGES/django.po (hunk 1 above)
Comment (ssorin) [reply src/dashboard/locale/fr/LC_MESSAGES/django.po:68]: ✅

File: src/dashboard/tests/dashboard/setup.py
Comment (ssorin) [reply src/dashboard/tests/dashboard/setup.py:1]: ✅

File: src/dashboard/apps/home/apps.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/dashboard/apps/home/apps.py:7]: Looks like it's not done :grimacing: :smile:

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**